cmd/root.go

```diff
@@ -0,0 +1,142 @@
+// Copyright © 2018 openSUSE opensuse-project@opensuse.org
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+
+	"github.com/openSUSE/helm-mirror/utils"
+
+	"github.com/openSUSE/helm-mirror/service"
+	"github.com/spf13/cobra"
+)
+
+var folder string
+var repoURL *url.URL
+var logger utils.Logger
+var verbose bool
+
+const rootDesc = `Mirror Helm Charts from an index file into a local folder.
+
+For example:
+
+helm-mirror https://yourorg.com/charts /yourorg/charts
+
+This will download the index file and the charts in to
+the folder indicated.
+
+The index file is a yaml that contains a list of
+charts in this format. Example:
+
+	apiVersion: v1
+	entries:
+	  chart:
+	  - apiVersion: 1.0.0
+	    created: 2018-08-08T00:00:00.00000000Z
+	    description: A Helm chart for your application
+	    digest: 3aa68d6cb66c14c1fcffc6dc6d0ad8a65b90b90c10f9f04125dc6fcaf8ef1b20
+	    name: chart
+	    urls:
+	    - https://kubernetes-charts.yourorganization.com/chart-1.0.0.tgz
+	  chart2:
+	  - apiVersion: 1.0.0
+	    created: 2018-08-08T00:00:00.00000000Z
+	    description: A Helm chart for your application
+	    digest: 7ae62d60b61c14c1fcffc6dc670e72e62b91b91c10f9f04125dc67cef2ef0b21
+	    name: chart
+	    urls:
+	    - https://kubernetes-charts.yourorganization.com/chart2-1.0.0.tgz
+
+This will download these charts
+
+	https://kubernetes-charts.yourorganization.com/chart-1.0.0.tgz
+	https://kubernetes-charts.yourorganization.com/chart2-1.0.0.tgz
+
+into your destination folder.`
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "mirror [Repo URL] [Destination Folder]",
+	Short: "Mirror Helm Charts from an index file into a local folder.",
+	Long:  rootDesc,
+	Args:  validateRootArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
i think here we also want to indent one more space to be in line with the other args